bitcoin/src/address.rs

```diff
@@ -643,9 +505,193 @@ impl NetworkValidation for NetworkUnchecked {
 /// representation of an address without the network validation tag, which is used to ensure that
 /// addresses are used only on the appropriate network.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-struct AddressInner {
-    payload: Payload,
-    network: Network,
+enum AddressInner {
+    P2pkh {
+        hash: PubkeyHash,
+        prefix: LegacyP2pkhPrefix,
+    },
+    P2sh {
+        hash: ScriptHash,
+        prefix: LegacyP2shPrefix,
+    },
+    Segwit {
+        version: WitnessVersion,
+        /// Guaranteed to be between 2 and 40 bytes.
+        program: PushBytesBuf,
+        bech32_hrp: Hrp,
+    },
+}
+
+impl AddressInner {
```

Comment:
Am I alone in prefering to not have any methods on the `AddressInner` type and just have them on `Address`? So that `Inner` is just a data structure? Like none of these methods are exposed and `Address` has each of them too with a proxy impl, which seems unnecessary.

Reply:
Yes, I like that idea. Will remove. Thanks man.